### ISSUE TITLE
CSI index for annotation track [gtf/bed] seems not supported

### ISSUE THREAD

**Issue by bioShaun** [igvteam/igv.js#1218]
with tbi index for gtf file
![tbi-index](https://user-images.githubusercontent.com/22510535/99023540-c4665280-259f-11eb-8cc6-b2d26cf25845.jpg)

with csi index
![csi-index](https://user-images.githubusercontent.com/22510535/99023565-cfb97e00-259f-11eb-832e-31d30dbd9f90.jpg)

and no error messages found.

**Comment by jrobinso** [comment]
It is supported, many people use it.   Can you zip up and post an example file, including the "config" part of your html?   Also state the version of igv.js you are using.

**Comment by bioShaun** [comment]
<div id="igv-div"></div>
    <script src="https://cdn.jsdelivr.net/npm/igv@2.7.0/dist/igv.min.js"></script>
    <script>
      var igvDiv = document.getElementById("igv-div");

      var dbBase = `http://localhost:3100/public/ref/oryza_sativa`;

      var annTrack = [
        {
          name: "Genes-tbi",
          format: "gtf",
          url: `${dbBase}/sorted.genes.gtf.gz`,
          indexURL: `${dbBase}/sorted.genes.gtf.gz.tbi`,
          order: 1000000,
        },
        {
          name: "Genes-csi",
          format: "gtf",
          url: `${dbBase}/sorted.genes.gtf.gz`,
          indexURL: `${dbBase}/sorted.genes.gtf.gz.csi`,
          order: 1000000,
        },
      ];

      console.log(annTrack);

      var options = {
        reference: {
          name: this.genome,
          fastaURL: `${dbBase}/sequences.fa`,
          indexURL: `${dbBase}/sequences.fa.fai`,
        },

        tracks: annTrack,
      };


      igv.createBrowser(igvDiv, options).then(function (browser) {
        console.log("Created IGV browser");
      });
    </script>`
`

**Comment by bioShaun** [comment]
![tbi-csi](https://user-images.githubusercontent.com/22510535/99025008-c251c300-25a2-11eb-8f71-095f1f8eccde.jpg)

this is the display

**Comment by jrobinso** [comment]
Thanks,  I'll check it out.   Your code looks fine.

**Comment by bioShaun** [comment]
thanks!

**Comment by bioShaun** [comment]
[example_file.tar.gz](https://github.com/igvteam/igv.js/files/5534593/example_file.tar.gz)
this is my test file

**Comment by jrobinso** [comment]
I can reproduce the issue with your config code.   It works for me without a visibility window set, but does not (as you observer) if visibility window is not set.   To be honest we always set a visibility window and I'm not sure this has been tried before.

Here's what works for me, and screenshot.   I will dig into this further tomorrow and try to figure out why it won't work without a visibility window.   It should, although you will load an entire chromosome of data which is usually not desired.

```
        {
            genome: "hg19",
            locus: "chr1:829,544-1,647,875",
            tracks:
                [
                    {
                        name: "Genes-csi",
                        format: "gtf",
                        url: `../test/data/tabix/sorted.genes.gtf.gz`,
                        indexURL: `../test/data/tabix/sorted.genes.gtf.gz.csi`,
                        order: 1000000,
                        visibilityWindow: 1000000
                    }
                ]
        }
```

<img width="1267" alt="Screen Shot 2020-11-12 at 8 27 47 PM" src="https://user-images.githubusercontent.com/933148/99028833-baa00900-2525-11eb-9a6d-c83c610bb5c8.png">

**Comment by bioShaun** [comment]
After add the visibilityWindow, it worked with the wheat genome, thanks a lot.
![wheat-fullzoom](https://user-images.githubusercontent.com/22510535/99037592-f2a75a80-25be-11eb-893e-9e8446251872.jpg)
![wheat-visiblezoom](https://user-images.githubusercontent.com/22510535/99037602-f89d3b80-25be-11eb-8afb-bbb9089ef298.jpg)

**Comment by jrobinso** [comment]
I just pushed a fix that will allow csi to be used without a visibility window.   You should still use one if your data density is high as without it data will be loaded by whole chromosome.    This fix will be in release 2.7.1,  in the next few days.
